connectors/s3-connector.js

```diff
@@ -18,7 +19,12 @@ async function uploadManifestToS3(parameters) {
       },
     });
 
-    const manifestCsv = convertObjectArrayToCsv(parameters.manifest);
+    const parsedManifest = JSON.parse(parameters.manifest);
+    if (!Array.isArray(parsedManifest) || !parsedManifest) {
```

Comment:
The order of the two condition should be swapped, to check for undefined/null first, then check type of the data.